src/NServiceBus.AcceptanceTests/Core/Causation/When_overriding_conversation_id_generation.cs

```diff
@@ -0,0 +1,96 @@
+﻿namespace NServiceBus.AcceptanceTests.Causation
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_overriding_conversation_id_generation : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_use_custom_id()
+        {
+            var myBusinessMessage = new MessageSentOutsideOfHandlerMatchingTheConvention
+            {
+                MyBusinessId = "some id"
+            };
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<CustomGeneratorEndpoint>(b => b.When(async session =>
+                {
+                    await session.SendLocal(myBusinessMessage);
+
+                    await session.SendLocal(new MessageSentOutsideOfHandlerNotMatchingTheConvention());
+                }))
+                .Done(c => c.MatchingMessageReceived && c.NonMatchingMessageReceived)
+                .Run();
+
+            Assert.AreEqual(myBusinessMessage.MyBusinessId, context.MatchingConversationIdReceived);
+            Assert.True(Guid.TryParse(context.NonMatchingConversationIdReceived, out var _));
+        }
+
+        public class Context : ScenarioContext
+        {
+            public string MatchingConversationIdReceived { get; set; }
+            public bool MatchingMessageReceived { get; set; }
+            public string NonMatchingConversationIdReceived { get; set; }
+            public bool NonMatchingMessageReceived { get; set; }
+        }
+
+        public class CustomGeneratorEndpoint : EndpointConfigurationBuilder
+        {
+            public CustomGeneratorEndpoint()
+            {
+                EndpointSetup<DefaultServer>(c => c.CustomConversationId(MyCustomConversationIdStrategy));
```

Comment:
> CustomConversationId

SetCustomConversationIdStrategy? SetCustomConversationIdGenerator? DefineCustomConversationIdStrategy?

Or is it clear enough based on the parameter you need to supply?

Reply:
Perhaps just `CustomConversationIdStrategy` ?